openedx/core/djangoapps/user_api/accounts/tests/test_retirement_views.py

```diff
@@ -0,0 +1,1492 @@
+        """
+        headers = build_jwt_headers(self.test_superuser)
+
+        mock_handler = mock.MagicMock()
+        mock_handler.side_effect = Exception("Tango")
+
+        try:
+            USER_RETIRE_MAILINGS.connect(mock_handler)
+
+            # User should still have 2 "True" subscriptions.
+            self.assert_status_and_tag_count(
+                headers,
+                expected_status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                expected_tag_value="True",
+                expected_content="Tango"
+            )
+        finally:
+            USER_RETIRE_MAILINGS.disconnect(mock_handler)
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Account APIs are only supported in LMS')
+class TestPartnerReportingCleanup(ModuleStoreTestCase):
+    """
+    Tests the partner reporting cleanup endpoint.
+    """
+
+    def setUp(self):
+        super(TestPartnerReportingCleanup, self).setUp()
+        self.test_superuser = SuperuserFactory()
+        self.course = CourseFactory()
+        self.course_awesome_org = CourseFactory(org='awesome_org')
+        self.headers = build_jwt_headers(self.test_superuser)
+        self.headers['content_type'] = "application/json"
+        self.url = reverse('accounts_retirement_partner_report')
+        self.maxDiff = None
+
+    def create_partner_reporting_statuses(self, is_being_processed=True, num=2):
+        """
+        Creates and returnes the given number of test users and UserRetirementPartnerReportingStatuses
+        with the given is_being_processed value.
+        """
+        statuses = []
+        for _ in range(num):
+            user = UserFactory()
+            reporting_status = UserRetirementPartnerReportingStatus.objects.create(
+                user=user,
+                original_username=user.username,
+                original_email=user.email,
+                original_name=user.first_name + ' ' + user.last_name,
+                is_being_processed=is_being_processed
+            )
+
+            statuses.append(reporting_status)
+
+        return statuses
+
+    def assert_status_and_count(self, statuses, remaining_count, expected_status=status.HTTP_204_NO_CONTENT):
+        """
+        Performs a test client DELETE against the retirement reporting cleanup endpoint. It generates
+        the JSON of usernames to clean up based on the given list of UserRetirementPartnerReportingStatuses,
+        asserts that the given number of UserRetirementPartnerReportingStatus rows are still in the database
+        after the operation, and asserts that the given expected_status HTTP status code is returned.
+        """
+        usernames = [{'original_username': u.original_username} for u in statuses]
+
+        data = json.dumps(usernames)
+        response = self.client.delete(self.url, data=data, **self.headers)
+        print(response)
+        print(response.content)
```

Comment:
Did you mean to leave these?

Reply:
I've been leaving the prints in these across the endpoint tests since pytest eats them unless there's a failure, and when there is a failure this is generally the information you'll need to debug it. I'm open to changing them to some other type of output, but this has been working for me.

Reply:
It's fine.